packages/dbos-compiler/cli.ts

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env node
+
+import { Command } from "commander";
+import path from 'node:path';
+import fs from 'node:fs';
+import fsp from 'node:fs/promises';
+import { generateCreate, generateDrop } from "./generator.js";
+import { parseConfigFile } from '@dbos-inc/dbos-sdk/dist/src/dbos-runtime/config.js'
+import * as pg from 'pg'
+import { type CompileResult, compile } from "./compiler.js";
+
+async function emitSqlFiles(outDir: string, result: CompileResult, appVersion?: string | boolean) {
+  await fsp.mkdir(outDir, { recursive: true });
+
+  const createFile = await fsp.open(path.join(outDir, "create.sql"), "w");
+  try {
+    const append = async (sql: string) => { await createFile.write(sql); };
+    await generateCreate(append, result, appVersion);
+  } finally {
+    await createFile.close();
+  }
+
+  const dropFile = await fsp.open(path.join(outDir, "drop.sql"), "w");
+  try {
```

Comment:
Why does the compilation step also drop the old versions? Shouldn't that be part of the deploy step?

Reply:
The compiler has two modes - `compile` and `deploy`. `compile` mode generates `create.sql` and `drop.sql`, `deploy` mode runs the contents of `create.sql` against the db directly (w/o actually writing the file to disk). For our cloud, my understanding is that we will cleanup old versions based on naming convention (when we clean up version 12345 of the app, drop all the routines that start with `v12345`) so the deploy mode doesn't need the drop capability. @maxdml should have more info on how we will eventually need to cleanup versioned JS SP routines

Reply:
Okay, so what's the expected user flow for using this?  Do you run `deploy` without running `compile` first? Is `compile` primarily for debugging purposes?

Reply:
no, they are independent. The results of `deploy` are the same as running the generated `create.sql` file. 

Again, I originally implemented the "generate SQL files" feature then later added the "update DB directly". I can remove it entirely if we want, but I'm thinking that it would be useful for debugging, local development and non DBOS-Cloud scenarios.

Reply:
Got it! Makes sense, and it's good to have `compile` around, just needs to be documented

Reply:
Will do. Obviously still need to add docs for this feature